### PR TITLE
Fix issues in the authentication flow of applications

### PR DESCRIPTION
**Pull request by Rashmini**
### Proposed changes in this pull request

Following issues will be fixed with this PR.

- SP_AUTH_SCRIPT table is not getting updated when the authentication flow is reverted back to default.
- Application is not getting deleted from the SP_AUTH_SCRIPT table, when the application is deleted.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 1]
```diff
@@ -4845,6 +4846,7 @@ public void deleteApplicationByResourceId(String resourceId,
 
         try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
             ServiceProvider application = getApplicationByResourceId(resourceId, tenantDomain);
+            int appIdUsingResourceId = getAppIdUsingResourceId(resourceId, tenantDomain);
```
Comment (dewniMW) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:4849]: can't we use `application.getApplicationID()` instead?

Comment (Rashmini) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with 43ba30a50e3dda59495ca8caba174e8259b9122c

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 2]
```diff
@@ -4858,6 +4860,7 @@ public void deleteApplicationByResourceId(String resourceId,
                     int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
                     deleteAppStatement.setInt(ApplicationTableColumns.TENANT_ID, tenantId);
                     deleteAppStatement.execute();
+                    deleteAuthenticationScript(applicationId, connection);
```
Comment (dewniMW) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:4863]:
```suggestion
                    deleteAuthenticationScript(application.getApplicationID(), connection);
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 3]
```diff
@@ -4847,6 +4848,7 @@ public void deleteApplicationByResourceId(String resourceId,
             ServiceProvider application = getApplicationByResourceId(resourceId, tenantDomain);
 
             if (application != null) {
+                int applicationId = application.getApplicationID();
```
Comment (dewniMW) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:4851]:
```suggestion
```

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 2 above)
Comment (Rashmini) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with 8a5c5dc87781e288f209a10ac9d79fd1e8027ec6

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 3 above)
Comment (Rashmini) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with 8a5c5dc87781e288f209a10ac9d79fd1e8027ec6

**Review by dewniMW — APPROVED** [review]

**Pull request merged**